server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -578,4 +580,24 @@ private static CircuitBreaker inFlightRequestsBreaker(CircuitBreakerService circ
         // We always obtain a fresh breaker to reflect changes to the breaker configuration.
         return circuitBreakerService.getBreaker(CircuitBreaker.IN_FLIGHT_REQUESTS);
     }
+
+    /**
+     * Perform an HTML escape operation on a String input to prevent XSS vulnerability.
+     * @param text the String to be escaped.
+     * @return The escaped result String.
+     */
+    private String escapeHtml(String text) {
+        StringBuilder out = new StringBuilder();
+        for (int i = 0; i < text.length(); i++) {
+            char c = text.charAt(i);
+            if (c == '"' || c == '\'' || c == '<' || c == '>' || c == '&') {
```

Comment:
[OWasp also recommends '/'](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-rules-summary)

I think that will conflict with the URI/route path though.

Reply:
Thanks for your review! 👍
Ah, I didn't realized that the slash "/" is also a character that recommended to deal with. 😂 But your thought is correct that it will make unreadable for the URI in the error message.
Frankly speaking, I even think my current code to do the HTML escape is over-engineering. URLEncode (which will escape the `<`, `>`, and `&` characters to `%3C`, `%3E`, and `%26`, respectively) is likely enough to satisfy the requirement, and there is native Java API to do that, but it's just not a professional solution for XSS problem (https://security.stackexchange.com/questions/8867/urlencode-vs-htmlencode-for-xss-prevention).

Reply:
Hi @reta, thanks for the opinion!
My reason for not using the additional library for sanitizing HTML is:
The purpose of the code change is to satisfy the security scanners so that they won't complain, instead of resolving actual vulnerability, so I don't think it worth to add a new Java dependency to the code base. Using additional library to the code will enlarge the file size for the build artifact, and adding maintenance burden for upgrading the library.

So I would rather using a simpler solution, to use a Java native API [URLEncoder](https://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html#encode-java.lang.String-java.lang.String-), `URLEncoder.encode(String s, String enc)`, which achieves the goal in the issue, but it's not a quite correct way to solve XSS problem.

Reply:
thanks @tlfeng, I think `URLEncoder.encode` won't fit here: `{"error":"no handler found for uri [/a/b/c] and method [DELETE]"}` would become something like `{"error":"no handler found for uri [%2Fa%2Fb%2Fc] and method [DELETE]"}`, not very readable.

What we are dealing with is actually the fact that any "garbage" could be sent to Opensearch, for example `{"error":"no handler found for uri [/a/b/c<script>alert(\"!\")</script>] and method [DELETE]"}` - this is not even valid URI. May be what we should do instead, is to distinguish those two issue:

 - `uri` is not valid: return `{"error":"invalid uri has been requested"}`
 - `uri` is valid but not found, in this case it is sufficient to include `uri.getPath()` only: return `{"error":"no handler found for uri [${uri.getPath()}] and method [DELETE]"}`

The funny part is that HTTP method is also vulnerable to XSS attacks:

```
 curl -ik -X '<script>alert("!")</script>' 'https://localhost:9200/a/b/c'

{"error":"Unexpected http method: <script>alert(\"!\")</script>","status":405}
```

We should probably follow the same approach, the list of HTTP methods is more or less stable, we could reply to the "garbage" something like: `{"error":"Unsupported http method","status":405}`

Reply:
@reta Thanks for your comprehensive solution! I think it's really reasonable. Instead of encoding HTML special characters, not showing the invalid user input sounds like a simple and effective solution. 😄 I will update the code based on it.